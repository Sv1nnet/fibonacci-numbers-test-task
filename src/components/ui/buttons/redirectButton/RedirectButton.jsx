import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Button } from 'react-materialize';

import './redirect-btn.scss';

const RedirectButton = ({ className, text, to }) => (
  <div className="redirect-btn-container">
    <Link className={`redirect-btn ${className}`} to={to}>
      <Button className="light-green darken-3">{text}</Button>
    </Link>
  </div>
);

RedirectButton.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
};

RedirectButton.defaultProps = {
  className: '',
};

export default RedirectButton;
