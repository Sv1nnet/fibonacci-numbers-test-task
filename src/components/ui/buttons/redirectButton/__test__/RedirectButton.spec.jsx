import React from 'react';
import {
  it,
  describe,
  expect,
} from '@jest/globals';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'materialize-css';
import toJson from 'enzyme-to-json';
import RedirectButton from '../RedirectButton';

configure({ adapter: new Adapter() });

describe('Testing RedirectButton component', () => {
  it('Should match snapshot with required props', () => {
    const component = shallow(<RedirectButton to="/test-to" text="test text" />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Should match snapshot with extra class', () => {
    const testClass = 'test-class';
    const testTo = '/test-to';
    const textTo = 'test text';
    const component = shallow(<RedirectButton className={testClass} to={testTo} text={textTo} />);

    expect(component.find('Link').prop('className').includes(testClass)).toBe(true);
    expect(component.find('Link').prop('to').includes(testTo)).toBe(true);
    expect(component.find('Button').props().children).toBe(textTo);
    expect(toJson(component)).toMatchSnapshot();
  });
});
