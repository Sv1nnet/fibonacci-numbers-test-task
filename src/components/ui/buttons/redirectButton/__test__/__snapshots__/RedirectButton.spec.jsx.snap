// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RedirectButton component Should match snapshot with extra class 1`] = `
<div
  className="redirect-btn-container"
>
  <Link
    className="redirect-btn test-class"
    to="/test-to"
  >
    <Button
      className="light-green darken-3"
      node="button"
    >
      test text
    </Button>
  </Link>
</div>
`;

exports[`Testing RedirectButton component Should match snapshot with required props 1`] = `
<div
  className="redirect-btn-container"
>
  <Link
    className="redirect-btn "
    to="/test-to"
  >
    <Button
      className="light-green darken-3"
      node="button"
    >
      test text
    </Button>
  </Link>
</div>
`;
